internal/cliwrapper/cliwrapper.go

```diff
@@ -86,9 +86,9 @@ func (p *cliWrapper) KillAndClean(containerName string) error {
 	cmdKill := p.getPodmanCmd("kill", containerName)
 	p.logger.Debugf("Killing with command %v", cmdKill.Args)
 	if err := cmdKill.Run(); err != nil {
-		p.logger.Warningf("failed to kill pod %s, probably the execution terminated earlier", containerName)
+		p.logger.Infof("failed to kill pod %s, probably the execution terminated earlier", containerName)
```

Comment:
This one's tough without knowing the full context. If this is routine and expected, maybe it should be downgraded to debug. Maybe there should be more analysis of the response to determine whether "execution terminated earlier" (which is probably not worth logging at all) or whether the attempt failed for some other reason (which might warrant a warning or even error).

Reply:
I think this is a good thing, so debug or no logging.
In reality it should check to see if the container is still running instead of blindly killing it.